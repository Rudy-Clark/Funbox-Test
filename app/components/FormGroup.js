/* eslint-disable no-return-assign */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const Form = styled.form`
  margin: 20px auto 12px auto;
  display: flex;
  justify-content: space-between;
  flex-flow: row nowrap;
  border: 1px solid #e3e3e3;
  padding: 0;
  height: 32px;
  width: 354px;
  align-items: center;
`;

const Input = styled.input`
  border: none;
  flex-basis: 95%;
  height: 32px;
  margin: 0 12px;
  outline: none;
`;

const Icon = styled.svg`
  width: 28px;
  height: 28px;
  padding: 3px;
`;

const Button = styled.button`
  border: none;
  border-left: 1px solid #e3e3e3;
  background-color: none;
  background: none;
  padding: 0;
  margin: 0;
  position: relative;
  flex-basis: 0%;
  height: 32px;
`;

const FormGroup = ({ addRoute }) => {
  let input;
  const onSubmit = e => {
    e.preventDefault();
    if (!input.value.trim()) return;
    addRoute(input.value);
    input.value = '';
  };
  return (
    <Form onSubmit={onSubmit}>
      <Input ref={node => (input = node)} id="suggest" type="text" />
      <Button type="submit">
        <Icon
          viewBox="0 0 42 42"
          style={{ enableBackground: 'new 0 0 42 42' }}
          x="0px"
          y="0px"
        >
          <polygon points="42,20 22,20 22,0 20,0 20,20 0,20 0,22 20,22 20,42 22,42 22,22 42,22" />
        </Icon>
      </Button>
    </Form>
  );
};

FormGroup.propTypes = {
  addRoute: PropTypes.func.isRequired,
};

export default FormGroup;
