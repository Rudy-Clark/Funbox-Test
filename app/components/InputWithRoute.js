import React from 'react';
import styled from 'styled-components';
import RouteList from '../containers/RouteList';
import Form from '../containers/Form';

const Wrapper = styled.div`
  flex-basis: 45%;
`;

const InputWithRoute = () => (
  <Wrapper>
    <Form />
    <RouteList />
  </Wrapper>
);

export default InputWithRoute;
