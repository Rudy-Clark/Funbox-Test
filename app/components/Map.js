import React from 'react';
import styled from 'styled-components';

const Map = styled.div`
  flex-basis: 45%;
  height: 320px;
`;

const MapView = () => <Map id="map" />;

export default MapView;
