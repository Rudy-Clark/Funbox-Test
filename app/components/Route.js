import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const LI = styled.li`
  margin: 0 0 7px;
  display: flex;
  flex-flow: row nowrap;
  align-items: start;
  cursor: pointer;
`;

const IconDelete = styled.svg`
  width: 26px;
  height: 22px;
  cursor: pointer;
  flex-basis: 5%;
`;

const Title = styled.p`
  flex-basis: 95%;
  line-height: 20px;
  padding: 2px 5px;
  background-color: #1282cb;
  border-radius: 1px;
  height: 22px;
  color: #fff;
  font-size: 0.975em;
  margin: 0;
  word-break: break-all;
  overflow: hidden;
`;

const Route = ({ routeName, deleteRoute }) => (
  <LI>
    <Title>{routeName}</Title>
    <IconDelete onClick={deleteRoute} viewBox="0 0 14.098 14.098">
      <path
        style={{ fill: 'rgb(237, 70, 82)' }}
        d="M13.198,0H0.901C0.404,0,0,0.403,0,0.9v12.299c0,0.496,0.404,0.9,0.901,0.9h12.298
			c0.497,0,0.9-0.404,0.9-0.9V0.901C14.098,0.403,13.694,0,13.198,0z M11.209,9.601l-1.608,1.607c-0.078,0.078-0.202,0.078-0.28,0
			L7.049,8.936l-2.272,2.273c-0.078,0.077-0.202,0.077-0.28,0L2.889,9.601c-0.077-0.078-0.077-0.202,0-0.28L5.16,7.049L2.888,4.777
			c-0.077-0.078-0.077-0.202,0-0.28l1.608-1.608c0.078-0.078,0.202-0.078,0.28,0l2.272,2.272L9.32,2.889
			c0.078-0.078,0.202-0.078,0.28,0l1.607,1.608c0.078,0.078,0.078,0.202,0,0.28l-2.27,2.272l2.272,2.272
			C11.286,9.399,11.286,9.523,11.209,9.601z"
      />
    </IconDelete>
  </LI>
);

Route.propTypes = {
  routeName: PropTypes.string.isRequired,
  deleteRoute: PropTypes.func.isRequired,
};

export default Route;
