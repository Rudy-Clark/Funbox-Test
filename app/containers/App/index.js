/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */

import React from 'react';
export default function App() {
  return (
    <div>
      <h1>Hello World</h1>
      <div id="maps" style={{ width: '100%', height: '650px' }} />
    </div>
  );
}
