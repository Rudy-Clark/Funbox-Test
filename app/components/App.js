import React from 'react';
import styled from 'styled-components';
import Map from './Map';
import InputWithRoute from './InputWithRoute';

const Container = styled.div`
  max-width: 980px;
  margin: 12px auto;
  display: flex;
  justify-content: space-around;
`;

const App = () => (
  <Container>
    <InputWithRoute />
    <Map />
  </Container>
);

export default App;
