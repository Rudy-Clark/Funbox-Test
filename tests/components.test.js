/* eslint-disable no-unused-vars */
import React from 'react';
import '@babel/polyfill';
import { mount, shallow } from 'enzyme';
import 'jest-styled-components';

import Route from '../app/components/Route';
import Map from '../app/components/Map';
import App from '../app/components/App';
import InputWithRoute from '../app/components/InputWithRoute';
import FormGroup from '../app/components/FormGroup';

describe('App', () => {
  it('App component snapshot', () => {
    const app = shallow(<App />);
    expect(app.exists()).toBe(true);
    expect(app).toMatchSnapshot();
  });
});

describe('InputWithRoute', () => {
  it('InputWithRoute snapshot', () => {
    const inputWithRoute = shallow(<InputWithRoute />);
    expect(inputWithRoute.exists()).toBe(true);
    expect(inputWithRoute).toMatchSnapshot();
  });
});

describe('Route', () => {
  const mock = {
    id: 'hiphop',
    routeName: 'test',
    deleteRoute: jest.fn(),
    isDragging: false,
  };
  const route = mount(<Route {...mock} />);
  const routerShallowRender = shallow(<Route {...mock} />);
  it('Route component snapshot with props', () => {
    expect(routerShallowRender).toMatchSnapshot();
  });

  it('check routeName text', () => {
    expect(route.text()).toEqual(mock.routeName);
  });

  it('check Route elements', () => {
    const cont = route.find('div');
    expect(cont.length).toBe(2);
    expect(cont.find('div div').length).toBe(1);
    expect(cont.find('svg').length).toBe(1);
    expect(cont.find('p').length).toBe(1);
  });

  it('check delete route', () => {
    route.find('svg').simulate('click');
    expect(mock.deleteRoute).toHaveBeenCalled();
  });

  it('Route contains props', () => {
    expect(route).toHaveProp({
      routeName: mock.routeName,
      isDragging: mock.isDragging,
      deleteRoute: mock.deleteRoute,
    });
  });
});

describe('FormGroup', () => {
  const mock = {
    requestSearch: jest.fn(),
    request: { error: false, loading: false, message: '' },
    resetError: jest.fn(),
  };

  const formGroup = shallow(<FormGroup {...mock} />);
  const fullFormGroup = mount(<FormGroup {...mock} />);

  it('FormGroup snapshot with props', () => {
    expect(formGroup.exists()).toBe(true);
    expect(formGroup).toMatchSnapshot();
  });

  it('FormGroup simulate click on empty form', () => {
    const form = fullFormGroup.find('form');
    form.simulate('submit');
    expect(mock.requestSearch).not.toHaveBeenCalled();
  });

  it('FormGroup check addRoute', () => {
    const value = 'test route';
    const form = fullFormGroup.find('form');
    form.find('input').instance().value = value;
    form.simulate('submit');
    expect(mock.requestSearch).toHaveBeenCalled();
    expect(mock.requestSearch.mock.calls.length).toBe(1);
    expect(mock.requestSearch.mock.calls[0][0]).toBe(value);
  });

  it('FormGroup check resetError', () => {
    const renderWithError = mount(
      <FormGroup
        {...mock}
        request={{ error: true, loading: false, message: '' }}
      />,
    );
    const errorBlock = renderWithError.find('div');
    errorBlock.simulate('click');
    expect(mock.resetError).toHaveBeenCalled();
    expect(mock.resetError.mock.calls.length).toBe(1);
    expect(fullFormGroup).toHaveProp({
      requestSearch: mock.requestSearch,
      request: { error: false, loading: false, message: '' },
      resetError: mock.resetError,
    });
  });
});

describe('Map', () => {
  it('Map snapshots', () => {
    const options = {
      center: [55.76, 37.64],
      zoom: 9,
      controls: [],
    };
    const map = shallow(<Map suggestView={'"suggest"'} options={options} />);
    expect(map.exists()).toBe(true);
    expect(map).toMatchSnapshot();
  });
});
