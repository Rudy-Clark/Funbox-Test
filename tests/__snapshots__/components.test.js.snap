// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App component snapshot 1`] = `
<div>
  <App__H1>
    Funbox Test
  </App__H1>
  <App__Container>
    <InputWithRoute />
    <MapView />
  </App__Container>
</div>
`;

exports[`InputWithRoute InputWithRoute snapshot 1`] = `
<InputWithRoute__Wrapper>
  <Connect(FormGroup) />
  <Connect(DragDropContext(ListRoutes)) />
</InputWithRoute__Wrapper>
`;

exports[`Map Map component snapshot 1`] = `
.c0 {
  -webkit-flex-basis: 45%;
  -ms-flex-preferred-size: 45%;
  flex-basis: 45%;
  height: 320px;
  width: 100%;
  margin: 20px 0 0;
}

@media (max-width:760px) {
  .c0 {
    -webkit-flex-basis: 85%;
    -ms-flex-preferred-size: 85%;
    flex-basis: 85%;
  }
}

<MapView>
  <Map
    id="map"
  >
    <StyledComponent
      forwardedClass={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Map-sc-16ezmnz-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "flex-basis:45%;height:320px;width:100%;margin:20px 0 0;@media (max-width:760px){flex-basis:85%;}",
            ],
          },
          "displayName": "Map",
          "render": [Function],
          "styledComponentId": "Map-sc-16ezmnz-0",
          "target": "div",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      id="map"
    >
      <div
        className="c0"
        id="map"
      />
    </StyledComponent>
  </Map>
</MapView>
`;

exports[`Route Route component snapshot 1`] = `
<Route__RouteCont>
  <Route__ROUTE>
    <Route__Title />
    <Route__IconDelete
      viewBox="0 0 14.098 14.098"
    >
      <path
        d="M13.198,0H0.901C0.404,0,0,0.403,0,0.9v12.299c0,0.496,0.404,0.9,0.901,0.9h12.298 c0.497,0,0.9-0.404,0.9-0.9V0.901C14.098,0.403,13.694,0,13.198,0z M11.209,9.601l-1.608,1.607c-0.078,0.078-0.202,0.078-0.28,0 L7.049,8.936l-2.272,2.273c-0.078,0.077-0.202,0.077-0.28,0L2.889,9.601c-0.077-0.078-0.077-0.202,0-0.28L5.16,7.049L2.888,4.777 c-0.077-0.078-0.077-0.202,0-0.28l1.608-1.608c0.078-0.078,0.202-0.078,0.28,0l2.272,2.272L9.32,2.889 c0.078-0.078,0.202-0.078,0.28,0l1.607,1.608c0.078,0.078,0.078,0.202,0,0.28l-2.27,2.272l2.272,2.272 C11.286,9.399,11.286,9.523,11.209,9.601z"
        style={
          Object {
            "fill": "rgb(237, 70, 82)",
          }
        }
      />
    </Route__IconDelete>
  </Route__ROUTE>
</Route__RouteCont>
`;

exports[`Route Route component snapshot with props 1`] = `
<Route__RouteCont>
  <Route__ROUTE
    isDragging={false}
  >
    <Route__Title>
      test
    </Route__Title>
    <Route__IconDelete
      onClick={[MockFunction]}
      viewBox="0 0 14.098 14.098"
    >
      <path
        d="M13.198,0H0.901C0.404,0,0,0.403,0,0.9v12.299c0,0.496,0.404,0.9,0.901,0.9h12.298 c0.497,0,0.9-0.404,0.9-0.9V0.901C14.098,0.403,13.694,0,13.198,0z M11.209,9.601l-1.608,1.607c-0.078,0.078-0.202,0.078-0.28,0 L7.049,8.936l-2.272,2.273c-0.078,0.077-0.202,0.077-0.28,0L2.889,9.601c-0.077-0.078-0.077-0.202,0-0.28L5.16,7.049L2.888,4.777 c-0.077-0.078-0.077-0.202,0-0.28l1.608-1.608c0.078-0.078,0.202-0.078,0.28,0l2.272,2.272L9.32,2.889 c0.078-0.078,0.202-0.078,0.28,0l1.607,1.608c0.078,0.078,0.078,0.202,0,0.28l-2.27,2.272l2.272,2.272 C11.286,9.399,11.286,9.523,11.209,9.601z"
        style={
          Object {
            "fill": "rgb(237, 70, 82)",
          }
        }
      />
    </Route__IconDelete>
  </Route__ROUTE>
</Route__RouteCont>
`;
